awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1512,43 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+    @property
+    def launched_by(self):
+        ancestor_job = self.ancestor_job
+
+        if ancestor_job.launch_type == "dependency":
+            return {
+                'id':None,
+                'name':'Generated by AWX',
+                'type':'Dependency',
+                'url':None
+            }
+
+        attr = {
+            "manual":"created_by",
+            "relaunch":"created_by",
+            "scheduled":"schedule" , 
+            "workflow":"workflow", 
+            "webhook":"job_template" , 
+            "sync":"project" ,
+            "scm":"inventory",}
+
+        obj = getattr(ancestor_job, attr.get(ancestor_job.launch_type, ''), None)
+        if obj is not None:
+            return {
+                'id':obj.id,
+                'name':getattr(obj, 'name', None) or obj.username,
+                'type':get_type_for_model(obj),
+                'url':obj.get_absolute_url()    
+            }
+        return {}
+
+    @property
+    def ancestor_job(self):
```

Comment:
In this function i have used ```get_workflow_job()``` function a lot. 
Let me know if i can reduce the number of function calls.

Reply:
cc @jbradberry

Reply:
This *could* get expensive given that we're nesting this call, and there's probably contexts where this is called in a list view.

Reply:
It we really wanted to know for sure, it might be useful to spawn ~50 workflow jobs, and then compare the time it takes to render `/api/v2/jobs/` *before* and *after* this change to make sure it's not _notably_ different.

Reply:
We could probably also optimize this a bit without having to do some many joins every time.  This is also a place where recursion might be a bit clearer than a loop.  I haven't tested this, but maybe something like:

```python
if self.spawned_by_workflow:
    return self.get_workflow_job().ancestor_job
else:
    return self
```

Reply:
I _think_ this works; I tested this by making a Workflow called "Grandpa", which has one node, another workflow called "Dad".  The Dad workflow has one node, which is a Job Template.

Grandpa --> Dad --> Demo Job Template

```python
In [5]: j.name
Out[5]: 'Demo Job Template'

In [6]: j.ancestor_job.name
Out[6]: 'Grandpa'

In [7]: j.get_workflow_job().name
Out[7]: 'Dad'
```

Reply:
Changed it to a one line recursive function.